task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -767,6 +768,14 @@ def _handle_request(self, msg: ToSupervisor, log: FilteringBoundLogger):
         if isinstance(msg, TaskState):
             self._terminal_state = msg.state
             self._task_end_time_monotonic = time.monotonic()
+        elif isinstance(msg, RuntimeCheckOnTask):
+            runtime_check_resp = self.client.task_instances.runtime_checks(id=self.id, msg=msg)
+            resp = runtime_check_resp.model_dump_json().encode()
+            if not runtime_check_resp.ok:
+                log.debug("Runtime checks failed on task %s, marking task as failed..", self.id)
+                self.client.task_instances.finish(
+                    id=self.id, state=TerminalTIState.FAILED, when=datetime.now(tz=timezone.utc)
+                )
```

Comment:
I'm not sure we should directly mark the task as failed here -- or at the least I think the task subprocess should be told about it so that it can run on_failure callbacks etc.

Reply:
Excellent point, we should send a message to the task runner to handle this portion, you are right.

Reply:
Handled in last batch of commits